pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -80,3 +87,25 @@ func updateTcTiDBAnnIfScale(tac *v1alpha1.TidbClusterAutoScaler) {
 	tac.Annotations[label.AnnTiDBLastAutoScalingTimestamp] = time.Now().String()
 	emptyAutoScalingCountAnn(tac, v1alpha1.TiDBMemberType)
 }
+
+func calculateTidbMetrics(tac *v1alpha1.TidbClusterAutoScaler, sts *appsv1.StatefulSet, client promClient.Client, instances []string) (int32, error) {
+	metric := calculate.FilterMetrics(tac.Spec.TiDB.Metrics)
+	mType, err := calculate.GenMetricType(tac, metric)
+	if err != nil {
+		return 0, err
+	}
+	switch mType {
+	case calculate.MetricTypeCPU:
+		return calculate.CalculateCpuMetrics(tac, sts, client, instances, metric, calculate.TidbSumCpuMetricsPattern, *tac.Spec.TiDB.MetricsTimeDuration, v1alpha1.TiDBMemberType)
+	default:
+		return 0, fmt.Errorf(calculate.InvalidTacMetricConfigureMsg, tac.Namespace, tac.Name)
```

Comment:
I recommend using `-1`, scaling to 0 is a useful feature and 0 should not be used to denote an error